module/zfs/vdev.c

```diff
@@ -4170,6 +4170,9 @@ vdev_clear(spa_t *spa, vdev_t *vd)
 	    vd->vdev_parent->vdev_ops == &vdev_spare_ops &&
 	    vd->vdev_parent->vdev_child[0] == vd)
 		vd->vdev_unspare = B_TRUE;
+
+	/* Clear recent events cache (i.e. duplicate events tracking) */
+	zfs_ereport_clear(spa, vd == spa->spa_root_vdev ? NULL : vd);
```

Comment:
if vdev_clear() is called on the root vdev, it calls itself recursively on all vdevs.  So passing vd=NULL seems not necessary for correctness, although maybe it performs better since subsequent calls (for child vdevs) would be iterating over a smaller list of events?  If that's the goal then it seems worthy of a comment.